From d807cbe0a7fb02ff6bda4f7be066c1323423743f Mon Sep 17 00:00:00 2001
From: Akseli Lahtinen <akselmo@akselmo.dev>
Date: Fri, 8 Dec 2023 12:25:18 +0000
Subject: [PATCH] Fix crash if window is closed when undoing

If dolphin is closed when undoing something,
this will crash. Set parentWidget as QPointer
to avoid crashing: Instead, the undo process
gets cancelled gracefully.

BUG:469586
---
 src/widgets/fileundomanager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/widgets/fileundomanager.cpp b/src/widgets/fileundomanager.cpp
index c1fef5db18..75bff39183 100644
--- a/src/widgets/fileundomanager.cpp
+++ b/src/widgets/fileundomanager.cpp
@@ -696,7 +696,7 @@ FileUndoManager::UiInterface *FileUndoManager::uiInterface() const
 class Q_DECL_HIDDEN FileUndoManager::UiInterface::UiInterfacePrivate
 {
 public:
-    QWidget *m_parentWidget = nullptr;
+    QPointer<QWidget> m_parentWidget;
     bool m_showProgressInfo = true;
 };
 
-- 
GitLab

